CONTRIBUTING.md

```diff
@@ -40,95 +40,134 @@ Merging Pull Requests (for project contributors with write access)
+> Note: The steps above would compile and generate the default output for azure-sdk-for-c witch includes static libraries only. See below section [Compiler Options](#compiler-options)
 
-```sh
+
+#### Visual Studio 2019
+Open project folder with Visual Studio. If VCPKG has been previously installed and set up like mentioned [above](#VCPKG). Everything will be ready to build.
+Right after opening project, Visual Studio will read cmake files and generate cache files automatically.
+
+### Linux / Mac
+VCPKG can also be used in Linux to avoid installing libraries globally. Follow instructions [here](#vcpkg) to use VCPKG in Linux.
+
+```bash
+# cd to project folder
+cd azure_sdk_for_c
+# create a new folder to generate cmake files for building (i.e. build)
+mkdir build
 cd build
-ctest -C Debug
+# generate files
+# cmake will automatically detect what C compiler is used by system by default and will generate files for it
+cmake ..
+# compile files. Cmake would call compiler and linker to generate libs
+make
 ```
 
-### Visual Studio 2019
-You can also build the project by simply opening the desired project directory in Visual Studio. Everything should be
-preconfigured to build and run tests.
+> Note: The steps above would compile and generate the default output for azure-sdk-for-c witch includes static libraries only. See below section [Compiler Options](#compiler-options)
+
+
+### Compiler Options
+By default, when building project with no options, next static libraries are generated
+- ``Libraries``:
+  - az_core
+  - az_iot
+  - az_keyvault
+  - az_storage_blobs
+- ``Platform Abstraction Layer``: Default empty implementation for platform functions like time and http stack. This default implementation is used to compile only but will return ERROR NOT IMPLEMENTED when running it.
+  - az_noplatform
+  - az_nohttp
+  - az_posix (on Lin/Mac)
+  - az_win32 (on Windows)
+- ``Samples``: By default, samples are built using the default PAL (see [running samples section](#running-samples)). This means that running samples would throw errors like:
+```bash
+./keys_client_example
+Running sample with no_op HTTP implementation.
+Recompile az_core with an HTTP client implementation like CURL to see sample sending network requests.
+
+i.e. cmake -DBUILD_CURL_TRANSPORT=ON ..
+```
+  - keys_client_example
+  - blobs_client_example
+
+When running cmake, next options can be used to change the output libraries/Pal/Samples:
+- `BUILD_CURL_TRANSPORT`: This option would build an HTTP transport library using CURL. It requires libcurl to be installed (vcpkg or globally). This option will make samples to be linked with this HTTP and be functional to send HTTP requests<br>
+use it as
+```bash
+cmake -DBUILD_CURL_TRANSPORT ..
+cmake --build .
+```
+- `UNIT_TESTING`: This option requires cmocka to be installed and it will generate unit tests for each project.<br>
+use it as
+```bash
+cmake -DUNIT_TESTING ..
```

Comment:
Shouldn't this be `cmake -DUNIT_TESTING=ON ..` ?
```suggestion
cmake -DUNIT_TESTING=ON ..
```

Reply:
https://github.com/Azure/azure-sdk-for-c/pull/434